fennel.lua

```diff
@@ -582,7 +594,8 @@ local function localMangling(str, scope, ast)
     end)
 
     local raw = mangling
-    while scope.unmanglings[mangling] do
+    while scope.unmanglings[mangling] or (allowedGlobals and
+                                          globalAllowed(mangling)) do
```

Comment:
Correct me if I'm wrong, but I think this will fail to account for the shadowing issue when `allowedGlobals` is false, because globals that exist as valid lua symbols aren't entered into the manglings table. The test you added on line 175 wouldn't account for this, since it's just using the default `allowedGlobals`

I suspect the only way to account for this with 100% reliability is to have mangling also add a prefix of some sort, similar to the way global manglings does. I don't know if that's something we want to avoid or not, but since you can't know what globals exist during runtime it's the only surefire approach. Not that it would be foolproof, but someone would at least have to work pretty hard to break it.

Another alternative, which would add fewer prefixes all over the place but still allow accidental shadowing when doing lua interop (or with globals created with `(tset _G foo bar)` would be to enter all globals created via `global` to the manglings/unmanglings table, whether they've been mangled or not.

Reply:
...it occurs to me that this is intentional - and that this approach still works so long as we're comfortable with `{ allowedGlobals = false }` means losing any protection against mangling causing accidental shadowing, since fennel can't know ahead of time what the globals will be.

If so, I suggest we make sure that's made clear in the docs/cli help.

Reply:
Yes, basically this means that the checking we do to ensure that locals and globals don't collide is contingent on not disabling strict local checking.

> enter all globals created via global to the manglings/unmanglings table, whether they've been mangled or not.

Interesting; I hadn't considered this approach, but it does allow us to apply the conflict checks more widely. I'll give it a shot tomorrow; thanks for the suggestion. Maybe the trade-off I made isn't strictly necessary.

Reply:
Cool; I think that did the trick! Thanks for the suggestion to add the globals to the manglings; that's a more thorough fix for the problem.